src/main/java/com/microsoftopentechnologies/windowsazurestorage/helper/CredentialMigration.java

```diff
@@ -0,0 +1,222 @@
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+/**
+ *
+ * @author arroyc
+ */
+public class CredentialMigration {
+    private static final Logger LOGGER = Logger.getLogger(CredentialMigration.class.getName());
+    
+    //private transient Log upgradelLogger = LogFactory.getLog(this.getClass());
+    public static List<StorageAccountInfo> getOldStorageConfig(String fileName) throws SAXException, IOException, ParserConfigurationException {
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder = factory.newDocumentBuilder();
+
+        // Load the legacy storage config XML document, parse it and return a list of storage accounts
+        Document document = builder.parse(new File(fileName));
+
+        List<StorageAccountInfo> storages = new ArrayList<StorageAccountInfo>();
+
+        NodeList nodeList = document.getElementsByTagName("com.microsoftopentechnologies.windowsazurestorage.beans.StorageAccountInfo");
+        for (int i = 0; i < nodeList.getLength(); i++) {
+            Node node = nodeList.item(i);
+
+            if (node.getNodeType() == Node.ELEMENT_NODE) {
+                Element elem = (Element) node;
+
+                // Get the value of all sub-elements.
+                String accName = elem.getElementsByTagName("storageAccName")
+                        .item(0).getChildNodes().item(0).getNodeValue();
+
+                String accKey = elem.getElementsByTagName("storageAccountKey").item(0)
+                        .getChildNodes().item(0).getNodeValue();
+
+                String blobURL = elem.getElementsByTagName("blobEndPointURL").item(0)
+                        .getChildNodes().item(0).getNodeValue();
+
+                storages.add(new StorageAccountInfo(accName, accKey, blobURL));
+            }
+        }
+
+        return storages;
+
+    }
+
+    public static void writeXMLElements(String jobConfigFile, Document document, NodeList nodeList, String storageCredId) throws TransformerException {
+        for (int i = 0; i < nodeList.getLength(); i++) {
+            Node node = nodeList.item(i);
+            Element credentialId = document.createElement("storageCredentialId");
+            credentialId.appendChild(document.createTextNode(storageCredId));
+            node.appendChild(credentialId);
+
+            // write the content into job config xml file
+            TransformerFactory transformerFactory = TransformerFactory.newInstance();
+            Transformer transformer = transformerFactory.newTransformer();
+            DOMSource source = new DOMSource(document);
+            StreamResult result = new StreamResult(new File(jobConfigFile));
+            transformer.transform(source, result);
+        }
+    }
+
+    public static void writeCredentialIdInJob(String jobName, String storageCredId) throws SAXException, IOException, ParserConfigurationException, TransformerConfigurationException, TransformerException {
```

Comment:
I wouldn't change the config files. The migration should just read the old storage accounts, but shouldn't patch jobs. The odds of something breaking are higher when you write raw bits in the config file